sound/soc/sof/pm.c

```diff
@@ -277,7 +277,7 @@ int snd_sof_resume(struct device *dev)
 	if (snd_sof_dsp_d0i3_on_suspend(sdev)) {
 		/* resume from D0I3 */
 		dev_dbg(sdev->dev, "DSP will exit from D0i3...\n");
-		ret = snd_sof_set_d0_substate(sdev, SOF_DSP_D0I0);
+		ret = snd_sof_set_dsp_power_state(sdev, SOF_DSP_D0I0);
```

Comment:
comment on line 292: I never realized it but the call to snd_sof_dsp_resume() is just awful. There is nothing telling the DSP that it needs to resume from D0i3. the same code is used in sof_resume()

It's really ugly to have the same function called at different levels to do different things. That should be a first cleanup.

Reply:
@plbossart snd_sof_dsp_resume() and snd_sof_dsp_suspend() deal with the distinction of whether we're resuming/suspending in/out of D0i3 or D3. Are you saying its ugly because at the top-level callback it looks like we're doing exactly the same thing? In that case, we could add the state arg just like we did for the suspend case to clarify what state we're resuming from to make it clear. Does that make sense?

Reply:
the point is that this code is moved anyways out of snd_sof_resume, so what I was suggesting here in what you ended-up implementing, but in later patches.